lapce-proxy/src/dispatch.rs

```diff
@@ -330,16 +330,17 @@ impl ProxyHandler for Dispatcher {
                                                 // (such as in minified javascript)
                                                 // Note that the start/end are column based, not absolute from the
                                                 // start of the file.
-                                                let display_range = mymatch
-                                                    .start()
-                                                    .saturating_sub(100)
-                                                    ..line
-                                                        .len()
-                                                        .min(mymatch.end() + 100);
                                                 line_matches.push((
                                                     lnum as usize,
                                                     (mymatch.start(), mymatch.end()),
-                                                    line[display_range].to_string(),
+                                                    line.chars()
```

Comment:
maybe it's better to start at the beginning of the match, rather than from the entire line?  You interate the whole string again

Reply:
The problem is that it's unclear what is the beginning of the match when it involves unicode characters that span multiple bytes since indexing works over bytes AFAIK.

Reply:
In addition, regex returns the start of the match in bytes, not characters (https://docs.rs/grep-matcher/0.1.5/grep_matcher/trait.Matcher.html#method.find)

Reply:
>just take a slice, and then chars, etc.

I don't think this is correct. If you take an arbitrary slice into a str as if its bytes, you will eventually slice into the middle of a wider character. What then? You can't know how to slice a str without first iterating over it. I've asked around and this is how checked substr implementation (`get`) does it too. If there is a way to slice over str, wouldn't `skip` of chars implement this already?

Regardless, we are only iterating over matching lines which is often quite short.